picamctl/picamctl.sh

```diff
@@ -0,0 +1,146 @@
+#!/bin/bash
+# akseidel 02/06/2021
+# A script automating the steps to run showmewencam's "camera-ctl".
+# The script will pause for continue or exit if run with an argument.
+
+# edit this next line to be the correct text pattern for your computer.
+portnamepat="tty.usbmodem" #typical for showmewebcam on Appl OS X
+#portnamepat="ttyACM" #probably typical for showmewebcam on Linux
+#portnamepat="tty."
+# The client is an application you want this script to also run. 
+runtheclient="true"
+clientname="Photo Booth"
+
+# initial cleanup
+initclean(){
+    reset
+    clear
+    # terminates all the ort screen sessions.
+    screen -ls | grep Attached | cut -d. -f1 | awk '{print $1}' | xargs kill
+    screen -ls | grep Detached | cut -d. -f1 | awk '{print $1}' | xargs kill
+}
+
+# run the client application
+runclient(){
+    if [ "$runtheclient" = "true" ]; then
+        # -g causes application to open in background 
+        # https://scriptingosx.com/2017/02/the-macos-open-command/
+        open -g -a "$clientname"
+    fi
+}
+
+# start the screen session
+doscreensession(){
+    echo ""
+    echo "Now establishing serial connection using 'screen' ..."
+    # For some reason screen stuffing the user, password and camera-ctl to
+    # a detached target screen does not work without the target screen having
+    # been attached at some point. Here the nest screen into a spawner screen
+    # trick is used to get around that.   
+    screen -dmS spawner
+    screen -S spawner -X screen screen -dR thispicam "$piusbwebcamport" 115200
+    sleep 0.3
```

Comment:
Why the delays? You can just read the output and wait for that.

Reply:
I don't know how to read the output. I hardly understand screens. Can you provide a reference link to start me on the right track to read screen output? The sleeps were necessary. Sometimes the Pi Zero was not ready in time. I thought it was lucky sleep would take decimal numbers instead of only integers. Perhaps by reading the screen output it might also be possible to detect the unrefreshed state camera-ctl can be in when the Pi Zero already had it running.

Reply:
Based on my tests with a Raspberry Pi 4, an even longer sleep is required, particularly this first one immediately after starting the screen connection.